### ISSUE TITLE
[Bug] qwen3_5 pre-fused gate_up load reads full tensors per host — degrades to ~19 MB/s mmap faults when ckpt > host RAM (397B)

### ISSUE THREAD

**Issue by MokusMokun** [sgl-project/sglang-jax#1445]
Found during #1419 bring-up (Qwen3.5-397B-A17B, v6e 8×8, tp64/dp4).

## Symptom

After `All weights loaded successfully`, the server appears to hang for hours before reaching the loader assert. The hang is misleading: all ranks sit at ~0% CPU with no open file descriptors on the checkpoint, because the reads are mmap page faults inside numpy conversion, not syscalls — nothing shows up in `lsof` or CPU profiles. 122B (250 GB checkpoint) never hits this; it only appears once the checkpoint exceeds host RAM.

## Root cause

`python/sgl_jax/srt/models/qwen3_5.py::_load_moe_gate_up` reads the FULL pre-fused `experts.gate_up_proj` tensor to host numpy via `_read_host` before slicing and uploading the per-host expert shards. Every host therefore materializes every layer's full `[E, 2*inter, hidden]` tensor even though it only keeps `E_local/E` of it.

At 397B the checkpoint (751 GB) is larger than host RAM (708 GB), so the page cache can never hold the working set: safetensors' mmap reads degrade to per-fault fetches through gcsfuse at ~19 MB/s, and the full-tensor reads turn into a multi-hour stall.

## Numbers

- Full read per host: 60 layers × `[512, 2048, 4096]` bf16 gate_up_proj ≈ **515 GB** (each layer's fused tensor is 512×2048×4096×2 B ≈ 8.6 GB).
- Checkpoint total **751 GB** > **708 GB** host RAM → page cache thrash, mmap faults served by gcsfuse at ~**19 MB/s** (measured), ≈ **7 h** projected load.
- With slice reads: per-host bytes drop to ~**64 GB** (E_local/E = 32/512 per host at 64-way expert sharding).

## Proposed fix (validated)

Read only each host's addressable expert slice inside the `jax.make_array_from_callback` callback, using safetensors lazy `get_slice` — the same idiom `weight_utils.py` already uses for its smart-slice paths (`_make_load_slice` ~L1142, `_smart_load_slice` ~L1236). A CPU test confirms the new path is bit-identical to the old one (bf16 and fp32 sources, including the transposed views).

<details>
<summary>Patch for <code>_load_moe_gate_up</code></summary>

```diff
     def _load_moe_gate_up(self, fm, weight_info, layer_idx):
-        """experts.gate_up_proj [E, 2*inter, hidden] -> w1/w3 [E, hidden, inter]."""
+        """experts.gate_up_proj [E, 2*inter, hidden] -> w1/w3 [E, hidden, inter].
+
+        Reads only this host's expert slice per shard (safetensors lazy slice)
+        instead of the full tensor: a full read is E/E_local x larger and at
+        397B scale (515 GB/host vs 64 GB) thrashes the page cache into
+        mmap-fault reads through gcsfuse (~19 MB/s -> 7 h load).
+        """
         src = f"model.language_model.layers.{layer_idx}.mlp.experts.gate_up_proj"
         block = self.language_model.model.layers[layer_idx].mlp
-        gu = self._read_host(fm, weight_info, src)  # [E, 2*inter, hidden]
-        inter = gu.shape[1] // 2
-        gate = gu[:, :inter, :]  # [E, inter, hidden]  (w1)
-        up = gu[:, inter:, :]  # [E, inter, hidden]   (w3)
-        w1 = np.transpose(gate, (0, 2, 1))  # [E, hidden, inter]
-        w3 = np.transpose(up, (0, 2, 1))
-        block.experts.w1.value = self._put(w1, (("data", "tensor"), None, None))
-        block.experts.w3.value = self._put(w3, (("data", "tensor"), None, None))
+        info = weight_info[src][0]
+        sl = fm.get_handle(info["file"]).get_slice(src)  # lazy [E, 2*inter, hidden]
+        num_experts, two_inter, hidden = sl.get_shape()
+        inter = two_inter // 2
+        sharding = NamedSharding(self.mesh, P(("data", "tensor"), None, None))
+
+        def _shard_reader(offset):
+            # idx is the global index tuple for one addressable shard of the
+            # [E, hidden, inter] target; only dim 0 (experts) is sharded.
+            def cb(idx):
+                e = idx[0]
+                chunk = np.asarray(sl[e, offset : offset + inter, :])  # [e, inter, hidden]
+                return np.transpose(chunk, (0, 2, 1))  # [e, hidden, inter]
+
+            return cb
+
+        shape = (num_experts, hidden, inter)
+        block.experts.w1.value = jax.make_array_from_callback(
+            shape, sharding, _shard_reader(0)
+        ).astype(self.dtype)
+        block.experts.w3.value = jax.make_array_from_callback(
+            shape, sharding, _shard_reader(inter)
+        ).astype(self.dtype)
```

</details>

## Validation

- 397B load+boot completes in minutes with the patch (plus a warm page cache), vs the ~7 h projection.
- Loader assert: 1038 consumed / 1886 skipped (visual + MTP) / 0 missing / 0 unexpected.
- GPQA-Diamond 5-seed median **0.894** vs published 0.884.

## Also proposed: `SGLANG_SKIP_GCSFUSE_WARMUP` env guard

`model_loader/loader.py::_warmup_safetensors_cache` re-reads the entire checkpoint on every server start — ~70–88 min for 807 GB — even when the page cache is already warm from a previous run. A one-line env guard (`SGLANG_SKIP_GCSFUSE_WARMUP=1` → skip warm-up) lets a warm restart boot in ~5.5 min. Default behavior is unchanged.

## Environment

- TPU v6e 8×8 (64 chips / 16 hosts), tp64/dp4, 16 hosts × 708 GB RAM.
- Checkpoint on gcsfuse-mounted GCS bucket.
- main@320fb93a.
